### ISSUE TITLE
Qt core dumps when RStudio launched with Qt Wayland

### ISSUE THREAD

**Issue by hermidalc** [rstudio/rstudio#4686]
### System details

    RStudio Edition : Desktop
    RStudio Version : 1.2.1335
    OS Version      : Linux (Fedora 29)
    R Version       : 3.5.1

### Steps to reproduce the problem

Set env var ```QT_QPA_PLATFORM=wayland``` and launch RStudio:
```bash
$ QT_QPA_PLATFORM=wayland rstudio
Warning: Ignoring XDG_SESSION_TYPE=wayland on Gnome. Use QT_QPA_PLATFORM=wayland to run on Wayland anyway.
qt.qpa.plugin: Could not load the Qt platform plugin "wayland" in "" even though it was found.
This application failed to start because no Qt platform plugin could be initialized. Reinstalling the application may fix this problem.

Available platform plugins are: eglfs, linuxfb, minimal, minimalegl, offscreen, vnc, wayland-egl, wayland, wayland-xcomposite-egl, wayland-xcomposite-glx, xcb.

Aborted (core dumped)
```

### Describe the problem in detail

When you launch RStudio 1.2.1335 in Linux on GNOME desktop (or KDE it doesn't matter) you always get this warning:
```
Warning: Ignoring XDG_SESSION_TYPE=wayland on Gnome. Use QT_QPA_PLATFORM=wayland to run on Wayland anyway.
```
If you attempt to set the env var mentioned to launch RStudio using native Wayland it fails and core dumps.  Not sure if this is a bug specific to RStudio or something with Qt and Wayland on Linux.

### Describe the behavior you expected

For RStudio to successfully launch using Wayland

**Comment by kevinushey** [comment]
What do you see if you run:

```
QT_DEBUG_PLUGINS=1 QT_QPA_PLATFORM=wayland /usr/lib/rstudio/bin/rstudio
```

I see some extra error output -- in particular:

```
Cannot load library /usr/lib/rstudio/plugins/platforms/libqwayland-generic.so: (libQt5WaylandClient.so.5: cannot open shared object file: No such file or directory)
```

and indeed RStudio does't bundle that library (`libQt5WaylandClient.so.5`), which may be the root cause of this issue.

**Comment by hermidalc** [comment]
> What do you see if you run:
> 
> ```
> QT_DEBUG_PLUGINS=1 QT_QPA_PLATFORM=wayland /usr/lib/rstudio/bin/rstudio
> ```
> 
> I see some extra error output -- in particular:
> 
> ```
> Cannot load library /usr/lib/rstudio/plugins/platforms/libqwayland-generic.so: (libQt5WaylandClient.so.5: cannot open shared object file: No such file or directory)
> ```
> 
> and indeed RStudio does't bundle that library (`libQt5WaylandClient.so.5`), which may be the root cause of this issue.

You probably need to install with your package manager qt5wayland or whatever it's called for Ubuntu.  In Fedora I already had it installed, so when I run ```QT_DEBUG_PLUGINS=1 QT_QPA_PLATFORM=wayland /usr/lib/rstudio/bin/rstudio``` I get a different error:
```
Cannot load library /usr/lib/rstudio/plugins/platforms/libqwayland-generic.so: (/lib64/libQt5WaylandClient.so.5: version `Qt_5_PRIVATE_API' not found (required by /usr/lib/rstudio/plugins/platforms/libqwayland-generic.so))
QLibraryPrivate::loadPlugin failed on "/usr/lib/rstudio/plugins/platforms/libqwayland-generic.so" : "Cannot load library /usr/lib/rstudio/plugins/platforms/libqwayland-generic.so: (/lib64/libQt5WaylandClient.so.5: version `Qt_5_PRIVATE_API' not found (required by /usr/lib/rstudio/plugins/platforms/libqwayland-generic.so))"
```
Why is RStudio referencing ```Qt_5_PRIVATE_API```?

**Comment by kevinushey** [comment]
What version of Qt do you have installed? We bundle Qt 5.12.1; perhaps it's not compatible with the version found in `/lib64`?

**Comment by hermidalc** [comment]
> What version of Qt do you have installed? We bundle Qt 5.12.1; perhaps it's not compatible with the version found in `/lib64`?

Ok that's what it is, because the latest Qt5 (and Qt5Wayland) version available for Fedora is 5.11.3

So solution is that RStudio needs to bundle Qt5Wayland libraries of the same version as their Qt.  You already have a pull request open to do that so thank you!
